src/traces/treemap/style.js

```diff
@@ -35,39 +35,65 @@ function styleOne(s, pt, trace, opts) {
     var ptNumber = cdi.i;
     var lineColor;
     var lineWidth;
-    var opacity;
-
-    var depthFade = function(n) {
-        var base = trace.marker.opacitybase;
-        var step = trace.marker.opacitystep;
-
-        return n === 0 ? base :
-            Math.max(0, Math.min(1, n * step));
-    };
+    var fillColor = cdi.color;
 
     if(hovered) {
         lineColor = trace._hovered.marker.line.color;
         lineWidth = trace._hovered.marker.line.width;
-        opacity = trace._hovered.marker.opacity;
     } else {
-        if(helpers.isHierarchyRoot(pt)) {
+        var isRoot = helpers.isHierarchyRoot(pt);
+
+        if(!trace._hasColorscale) {
+            if(pt.onPathbar) {
+                // Combining colors with the background color.
+                // Better not to have transparent segments.
+                // Also this helps pathbar texts appear nicely below previous segments.
+                fillColor = Color.combine(Color.addOpacity(trace._backgroundColor, 1 - trace.pathbar.opacity), fillColor);
```

Comment:
Hmm. This logic is ok, but we can't use `pathbar.opacity` for this. Opacity attributes should set some opacity DOM attribute (in SVG traces at least :smirk: )

So, we should either:

- come up with another attribute name to replace `pathbar.opacity`
- or, drop `pathbar.opacity` altogether

At this stage, I would personally vote for dropping `pathbar.opacity`. If some users still want to have some opacity in their pathbar, they could always set an `rgba` color.

Reply:
OK. Let's drop that.

Reply:
Dropped in https://github.com/plotly/plotly.js/pull/4242/commits/3b104b143050a6bbeb0853486dd2ac5a3a82802c.